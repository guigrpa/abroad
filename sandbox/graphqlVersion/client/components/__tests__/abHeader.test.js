/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { Hints, Floats } from 'giu';
import Header from '../abHeader';
import $ from './jestQuery';

// https://github.com/facebook/react/issues/7386#issuecomment-238091398
jest.mock('react-dom');

describe('Header', () => {
  it('01 normal', () => {
    const tree = renderer
      .create(
        <div>
          <Floats />
          <Header onShowSettings={() => {}} scopes={[]} />
        </div>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('02 shows hints when the info button is clicked', () => {
    const component = renderer.create(
      <div>
        <Hints />
        <Floats />
        <Header onShowSettings={() => {}} scopes={[]} />
      </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    const infoBtn = $(tree, '.fa-question-circle');
    expect(infoBtn).not.toBeNull();
    infoBtn.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
