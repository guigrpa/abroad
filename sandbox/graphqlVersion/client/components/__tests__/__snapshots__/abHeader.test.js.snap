// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 01 normal 1`] = `
<div>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    style={
      Object {
        "WebkitFlex": "0 0 2.5em",
        "alignItems": "center",
        "backgroundColor": "#dcd6ff",
        "display": "flex",
        "flex": "0 0 2.5em",
        "flexDirection": "row",
        "fontSize": "1rem",
        "padding": "5px 4px",
      }
    }
  >
    <div>
      <span
        className="giu-select-custom"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <input
          onBlur={[Function]}
          onCopy={[Function]}
          onCut={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          style={
            Object {
              "cursor": "default",
              "height": 0,
              "left": 8,
              "opacity": 0,
              "padding": 0,
              "pointerEvents": "none",
              "position": "fixed",
              "top": 8,
              "width": 0,
              "zIndex": -80,
            }
          }
        />
        <div
          className="giu-drop-down-menu-title"
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "color": undefined,
              "cursor": "pointer",
              "display": "inline-block",
            }
          }
        >
          <div
            style={
              Object {
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            <i
              className="giu-icon fa fa-filter"
              id="madyMenuFilter"
              style={
                Object {
                  "color": undefined,
                  "cursor": "pointer",
                  "fontSize": undefined,
                  "letterSpacing": "normal",
                }
              }
              title="Filter"
            />
          </div>
        </div>
      </span>
    </div>
    <div
      style={
        Object {
          "margin": "0 1em",
        }
      }
    >
      <input
        className="giu-text-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Quick find"
        style={
          Object {
            "backgroundColor": "white",
            "border": "1px solid #999999",
            "color": "inherit",
            "fontFamily": "inherit",
            "fontSize": "0.8rem",
            "fontWeight": "inherit",
          }
        }
        type="text"
        value=""
      />
    </div>
    <div
      style={
        Object {
          "WebkitFlex": 1,
          "flex": 1,
        }
      }
    />
    <div
      style={
        Object {
          "fontWeight": 900,
          "letterSpacing": 3,
          "paddingBottom": 1,
        }
      }
    >
      MADY
      <i
        className="giu-icon fa fa-cog"
        id="madyBtnSettings"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
            "margin": "0 1em",
          }
        }
        title="Settings"
      />
    </div>
    <div
      style={
        Object {
          "marginRight": 10,
          "textAlign": "right",
        }
      }
    >
      <i
        className="giu-icon fa fa-question-circle"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
          }
        }
      />
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div>
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div style="-webkit-flex:0 0 2.5em;align-items:center;background-color:#dcd6ff;display:flex;flex:0 0 2.5em;flex-direction:row;font-size:1rem;padding:5px 4px;">
    <div>
      <span
        class="giu-select-custom"
        style="display:inline-block;position:relative;"
      >
        <input style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
        <div
          class="giu-drop-down-menu-title"
          style="cursor:pointer;display:inline-block;"
        >
          <div style="padding-left:10px;padding-right:10px;">
            <i
              class="giu-icon fa fa-filter"
              id="madyMenuFilter"
              style="cursor:pointer;letter-spacing:normal;"
              title="Filter"
            >
            </i>
          </div>
        </div>
      </span>
    </div>
    <div style="margin:0 1em;">
      <input
        class="giu-text-input"
        placeholder="Quick find"
        style="background-color:white;border:1px solid #999999;color:inherit;font-family:inherit;font-size:0.8rem;font-weight:inherit;"
        type="text"
        value=""
      >
    </div>
    <div style="-webkit-flex:1;flex:1;"></div>
    <div style="font-weight:900;letter-spacing:3px;padding-bottom:1px;">
      MADY
      <i
        class="giu-icon fa fa-cog"
        id="madyBtnSettings"
        style="cursor:pointer;letter-spacing:normal;margin:0 1em;"
        title="Settings"
      >
      </i>
    </div>
    <div style="margin-right:10px;text-align:right;">
      <i
        class="giu-icon fa fa-question-circle"
        style="cursor:pointer;letter-spacing:normal;"
      >
      </i>
    </div>
  </div>
</div>
`;

exports[`Header 02 shows hints when the info button is clicked 1`] = `
<div>
  <div
    className="giu-hints"
  />
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    style={
      Object {
        "WebkitFlex": "0 0 2.5em",
        "alignItems": "center",
        "backgroundColor": "#dcd6ff",
        "display": "flex",
        "flex": "0 0 2.5em",
        "flexDirection": "row",
        "fontSize": "1rem",
        "padding": "5px 4px",
      }
    }
  >
    <div>
      <span
        className="giu-select-custom"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <input
          onBlur={[Function]}
          onCopy={[Function]}
          onCut={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          style={
            Object {
              "cursor": "default",
              "height": 0,
              "left": 8,
              "opacity": 0,
              "padding": 0,
              "pointerEvents": "none",
              "position": "fixed",
              "top": 8,
              "width": 0,
              "zIndex": -80,
            }
          }
        />
        <div
          className="giu-drop-down-menu-title"
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "color": undefined,
              "cursor": "pointer",
              "display": "inline-block",
            }
          }
        >
          <div
            style={
              Object {
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            <i
              className="giu-icon fa fa-filter"
              id="madyMenuFilter"
              style={
                Object {
                  "color": undefined,
                  "cursor": "pointer",
                  "fontSize": undefined,
                  "letterSpacing": "normal",
                }
              }
              title="Filter"
            />
          </div>
        </div>
      </span>
    </div>
    <div
      style={
        Object {
          "margin": "0 1em",
        }
      }
    >
      <input
        className="giu-text-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Quick find"
        style={
          Object {
            "backgroundColor": "white",
            "border": "1px solid #999999",
            "color": "inherit",
            "fontFamily": "inherit",
            "fontSize": "0.8rem",
            "fontWeight": "inherit",
          }
        }
        type="text"
        value=""
      />
    </div>
    <div
      style={
        Object {
          "WebkitFlex": 1,
          "flex": 1,
        }
      }
    />
    <div
      style={
        Object {
          "fontWeight": 900,
          "letterSpacing": 3,
          "paddingBottom": 1,
        }
      }
    >
      MADY
      <i
        className="giu-icon fa fa-cog"
        id="madyBtnSettings"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
            "margin": "0 1em",
          }
        }
        title="Settings"
      />
    </div>
    <div
      style={
        Object {
          "marginRight": 10,
          "textAlign": "right",
        }
      }
    >
      <i
        className="giu-icon fa fa-question-circle"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
          }
        }
      />
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div>
  <div class="giu-hints"></div>
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div style="-webkit-flex:0 0 2.5em;align-items:center;background-color:#dcd6ff;display:flex;flex:0 0 2.5em;flex-direction:row;font-size:1rem;padding:5px 4px;">
    <div>
      <span
        class="giu-select-custom"
        style="display:inline-block;position:relative;"
      >
        <input style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
        <div
          class="giu-drop-down-menu-title"
          style="cursor:pointer;display:inline-block;"
        >
          <div style="padding-left:10px;padding-right:10px;">
            <i
              class="giu-icon fa fa-filter"
              id="madyMenuFilter"
              style="cursor:pointer;letter-spacing:normal;"
              title="Filter"
            >
            </i>
          </div>
        </div>
      </span>
    </div>
    <div style="margin:0 1em;">
      <input
        class="giu-text-input"
        placeholder="Quick find"
        style="background-color:white;border:1px solid #999999;color:inherit;font-family:inherit;font-size:0.8rem;font-weight:inherit;"
        type="text"
        value=""
      >
    </div>
    <div style="-webkit-flex:1;flex:1;"></div>
    <div style="font-weight:900;letter-spacing:3px;padding-bottom:1px;">
      MADY
      <i
        class="giu-icon fa fa-cog"
        id="madyBtnSettings"
        style="cursor:pointer;letter-spacing:normal;margin:0 1em;"
        title="Settings"
      >
      </i>
    </div>
    <div style="margin-right:10px;text-align:right;">
      <i
        class="giu-icon fa fa-question-circle"
        style="cursor:pointer;letter-spacing:normal;"
      >
      </i>
    </div>
  </div>
</div>
`;

exports[`Header 02 shows hints when the info button is clicked 2`] = `
<div>
  <div
    className="giu-hints"
  >
    <div
      className="giu-hint-screen"
      onClick={[Function]}
      style={
        Object {
          "height": "100vh",
          "left": 0,
          "position": "fixed",
          "top": 0,
          "width": "100vw",
          "zIndex": 50,
        }
      }
    >
      <div
        className="giu-backdrop"
        onTouchMove={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "cursor": "pointer",
            "height": "100vh",
            "left": 0,
            "position": "fixed",
            "top": 0,
            "width": "100vw",
          }
        }
      />
      <div
        style={
          Object {
            "color": "white",
            "fontFamily": "\\"Gloria Hallelujah\\", sans-serif",
            "fontSize": 20,
          }
        }
      >
        <div
          style={
            Object {
              "border": "1px solid white",
              "borderRadius": 8,
              "bottom": 20,
              "cursor": "pointer",
              "padding": "5px 25px",
              "position": "fixed",
              "right": 20,
            }
          }
        >
          Got it!
        </div>
      </div>
    </div>
  </div>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    }
  />
  <div
    style={
      Object {
        "WebkitFlex": "0 0 2.5em",
        "alignItems": "center",
        "backgroundColor": "#dcd6ff",
        "display": "flex",
        "flex": "0 0 2.5em",
        "flexDirection": "row",
        "fontSize": "1rem",
        "padding": "5px 4px",
      }
    }
  >
    <div>
      <span
        className="giu-select-custom"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <input
          onBlur={[Function]}
          onCopy={[Function]}
          onCut={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          style={
            Object {
              "cursor": "default",
              "height": 0,
              "left": 8,
              "opacity": 0,
              "padding": 0,
              "pointerEvents": "none",
              "position": "fixed",
              "top": 8,
              "width": 0,
              "zIndex": -80,
            }
          }
        />
        <div
          className="giu-drop-down-menu-title"
          onMouseDown={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "color": undefined,
              "cursor": "pointer",
              "display": "inline-block",
            }
          }
        >
          <div
            style={
              Object {
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            <i
              className="giu-icon fa fa-filter"
              id="madyMenuFilter"
              style={
                Object {
                  "color": undefined,
                  "cursor": "pointer",
                  "fontSize": undefined,
                  "letterSpacing": "normal",
                }
              }
              title="Filter"
            />
          </div>
        </div>
      </span>
    </div>
    <div
      style={
        Object {
          "margin": "0 1em",
        }
      }
    >
      <input
        className="giu-text-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Quick find"
        style={
          Object {
            "backgroundColor": "white",
            "border": "1px solid #999999",
            "color": "inherit",
            "fontFamily": "inherit",
            "fontSize": "0.8rem",
            "fontWeight": "inherit",
          }
        }
        type="text"
        value=""
      />
    </div>
    <div
      style={
        Object {
          "WebkitFlex": 1,
          "flex": 1,
        }
      }
    />
    <div
      style={
        Object {
          "fontWeight": 900,
          "letterSpacing": 3,
          "paddingBottom": 1,
        }
      }
    >
      MADY
      <i
        className="giu-icon fa fa-cog"
        id="madyBtnSettings"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
            "margin": "0 1em",
          }
        }
        title="Settings"
      />
    </div>
    <div
      style={
        Object {
          "marginRight": 10,
          "textAlign": "right",
        }
      }
    >
      <i
        className="giu-icon fa fa-question-circle"
        onClick={[Function]}
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
          }
        }
      />
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div>
  <div class="giu-hints">
    <div
      class="giu-hint-screen"
      style="height:100vh;left:0;position:fixed;top:0;width:100vw;z-index:50;"
    >
      <div
        class="giu-backdrop"
        style="background-color:black;cursor:pointer;height:100vh;left:0;position:fixed;top:0;width:100vw;"
      >
      </div>
      <div style="color:white;font-family:\\"Gloria Hallelujah\\", sans-serif;font-size:20px;">
        <div style="border:1px solid white;border-radius:8px;bottom:20px;cursor:pointer;padding:5px 25px;position:fixed;right:20px;">
          Got it!
        </div>
      </div>
    </div>
  </div>
  <div
    class="giu-floats"
    style="height:0;left:0;position:absolute;top:0;width:0;"
  >
  </div>
  <div style="-webkit-flex:0 0 2.5em;align-items:center;background-color:#dcd6ff;display:flex;flex:0 0 2.5em;flex-direction:row;font-size:1rem;padding:5px 4px;">
    <div>
      <span
        class="giu-select-custom"
        style="display:inline-block;position:relative;"
      >
        <input style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
        <div
          class="giu-drop-down-menu-title"
          style="cursor:pointer;display:inline-block;"
        >
          <div style="padding-left:10px;padding-right:10px;">
            <i
              class="giu-icon fa fa-filter"
              id="madyMenuFilter"
              style="cursor:pointer;letter-spacing:normal;"
              title="Filter"
            >
            </i>
          </div>
        </div>
      </span>
    </div>
    <div style="margin:0 1em;">
      <input
        class="giu-text-input"
        placeholder="Quick find"
        style="background-color:white;border:1px solid #999999;color:inherit;font-family:inherit;font-size:0.8rem;font-weight:inherit;"
        type="text"
        value=""
      >
    </div>
    <div style="-webkit-flex:1;flex:1;"></div>
    <div style="font-weight:900;letter-spacing:3px;padding-bottom:1px;">
      MADY
      <i
        class="giu-icon fa fa-cog"
        id="madyBtnSettings"
        style="cursor:pointer;letter-spacing:normal;margin:0 1em;"
        title="Settings"
      >
      </i>
    </div>
    <div style="margin-right:10px;text-align:right;">
      <i
        class="giu-icon fa fa-question-circle"
        style="cursor:pointer;letter-spacing:normal;"
      >
      </i>
    </div>
  </div>
</div>
`;
